src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -743,13 +823,17 @@ int fmi3_xml_set_attr_array(fmi3_xml_parser_context_t *context, fmi3_xml_elm_enu
     }
 
     /* write all attributes to array of correct type */
-    if (fmi3_xml_str_to_array_floatXX(context, str, arrPtr, arrSize, primType)) {
+    if (fmi3_base_type_enu_is_float(primType->baseType) && fmi3_xml_str_to_array_floatXX(context, str, arrPtr, arrSize, primType)) {
```

Comment:
I suggest you write it like:

```
if (fmi3_base_type_enu_is_float(primType->baseType)) {
   if (fmi3_xml_str_to_array_floatXX(...)) {
        ...
   }
} else if { ...
```
Otherwise you'll also try to parse as intXX even if it succeeded for floatXX. I mean it try to eval the else-if branch.

Reply:
Let us also discuss this piece tomorrow.